src/server.js

```diff
@@ -215,5 +224,91 @@ app.get(
   }
 );
 
+app.get(
+  `${PREFIX_API}/:${URL_PARAM_ADDRESS}/geocode`,
+  async (request, response) => {
+    let address = request.params[URL_PARAM_ADDRESS];
+
+    let geocodeRequest =
+      "https://maps.googleapis.com/maps/api/geocode/json?address=" +
+      address +
+      "&key=" +
+      env.API_KEY;
+
+    try {
+      const geocodeResponse = await (await fetch(geocodeRequest)).json();
+      const geocodeResponseStatus = geocodeResponse.status;
+
+      if (geocodeResponseStatus !== "OK") {
+        console.error(
+          "Geocoding error occured. Api response status: " +
+            geocodeResponseStatus
+        );
+        response
+          .status(500)
+          .json({ status: 500, error: { type: geocodeResponseStatus } });
+      } else {
+        response.json({
+          status: 200,
+          data: geocodeResponse.results[0].geometry.location,
+        });
+      }
+    } catch (err) {
+      console.error(err);
+      response
+        .status(500)
+        .json({ status: 500, error: { type: ERROR_GEOCODING_FAILED } });
+    }
+  }
+);
+
+/** Some Hard-coded data to display as restaurants temporarily. */
+function getRestaurants() {
+  const restaurant1 = {
+    name: "Nice Cafe",
+    address: "123 N. Street st.",
+    latLong: { lat: 37.0, lng: -122.0 },
+    rating: 5,
+    priceLevel: 2,
+    openingHours: "8:00am - 10:00pm",
+  };
+
+  const restaurant2 = {
+    name: "Bad Cafe",
+    address: "987 S. Street st.",
+    latLong: { lat: -37.0, lng: 122.0 },
+    rating: 2,
+    priceLevel: 4,
+    openingHours: "12:00am - 9:00am",
+  };
+
+  return [restaurant1, restaurant2];
+}
+
+app.get(
+  `${PREFIX_API}/:${URL_PARAM_EVENT_ID}/restaurants`,
+  async (request, response) => {
+    const restaurants = getRestaurants();
+    response.json({
+      status: 200,
+      data: Object.values(restaurants),
+    });
+  }
+);
+
 const port = 8080;
-app.listen(port, () => console.log(`Server listening on port: ${port}`));
+const server = app.listen(port, () =>
+  console.log(`Server listening on port: ${port}`)
+);
+
+//Attach Socket.io to the existing express server.
+const io = require("socket.io")(server);
+
+//When socket.io is connected to the server, we can listen for events.
+io.on("connection", (socket) => {
+  //This 'data submitted' message, recieved by the server, gets emitted by a client every time new user information is successfully submitted.
+  //In response, the server passes this refresh message and the event ID to let clients with the same ID know to refresh their search results.
+  socket.on("data submitted", (eventID) => {
+    io.emit("refresh", eventID);
```

Comment:
Thanks for the great documentation explaining what this block does.

However, it seems this will emit the message to all clients, which I think will cause a couple issues:

- you're sending out the eventID to all clients. A malicious client now knows your ID, and can hop into your meeting, and start messing with it
- each client will have to process every emission from all other events, which, if you have many simultaneous sessions going on, is a lot of unnecessary computation, and could slow down your client experience at scale (imagine you have 10,000 sessions at one time, and each is updated every 10 seconds- that means each client on average has to process 1,000 emissions from socket.io per second...)

I'm not familiar with socket.io, but I think you should be able to specify server side to whom you want to emit the message. E.g.:

// sending to all clients in 'game' room, including sender
  io.in('game').emit('big-announcement', 'the game will start soon');

// sending to individual socketid (private message)
  io.to(socketId).emit('hey', 'I just met you');

Reference: https://socket.io/docs/emit-cheatsheet/

Reply:
I updated to add clients to a room based on their event ID. When a user submits information, the event ID is still sent to the server, but the server can use that ID to send messages only to clients in the same session. The server no longer sends event ID's to the client. 

Thank you for pointing this out and let me know if my solution still poses an issue.